google/cloud/aiplatform/initializer.py

```diff
@@ -14,28 +14,142 @@
+        """Default project."""
+        if self._project:
+            return self._project
+
+        _, project_id = google.auth.default()
+        return project_id
+
+    @property
+    def location(self) -> str:
+        """Default location."""
+        return self._location if self._location else utils.DEFAULT_REGION
+
+    @property
+    def experiment(self) -> Optional[str]:
+        """Default experiment, if provided."""
+        return self._experiment
+
+    @property
+    def staging_bucket(self) -> Optional[str]:
+        """Default staging bucket, if provided."""
+        return self._staging_bucket
+
+    @property
+    def credentials(self) -> Optional[auth_credentials.Credentials]:
+        """Default credentials, if provided."""
+        return self._credentials
+
+    def get_client_options(
+        self,
+        location_override: Optional[str] = None,
+        prediction_client: Optional[bool] = False,
+    ) -> Dict[str, str]:
+        """Creates GAPIC client_options using location and type.
+
+        Args:
+            location_override (str):
+                Set this parameter to get client options for a location different from
+                location set by initializer. Must be a GCP region supported by AI
+                Platform (Unified).
+
+            prediction_client (bool):
+                True if service client is a PredictionServiceClient, otherwise defaults 
+                to False. This is used to provide a prediction-specific API endpoint.
+
+        Returns:
+            clients_options (dict):
+                A dictionary containing client_options with one key, for example
+                { "api_endpoint": "us-central1-aiplatform.googleapis.com" } or
+                { "api_endpoint": "asia-east1-prediction-aiplatform.googleapis.com" }
+        """
+        if not (self.location or location_override):
+            raise ValueError(
+                "No location found. Provide or initialize SDK with a location."
+            )
+
+        region = self.location if not location_override else location_override
+        region = region.lower()
+        prediction = "prediction-" if prediction_client else ""
+
+        utils.validate_region(region)
+
+        client_options = {
+            "api_endpoint": f"{region}-{prediction}{utils.PROD_API_ENDPOINT}"
+        }
+
+        return client_options
 
-	def __init__(self):
-		self.project = None
-		self.experiment = None
-		self.location = None
-		self.staging_bucket = None
-
-	def init(self, project: Optional[str] = None, location: Optional[str] = None,
-		experiment: Optional[str] = None, staging_bucket: Optional[str] = None):
-		if project:
-			self.project=project
-		if location:
-			self.location=location
-		if experiment:
-			self.experiment=experiment
-		if staging_bucket:
-			self.staging_bucket=staging_bucket
 
 # singleton to store init parameters: ie, aiplatform.init(project=..., location=...)
 singleton = Init()
```

Comment:
Consider using some class-level patterns to create the singleton (for example, to avoid user incorrectly using the `Init` class): https://python-3-patterns-idioms-test.readthedocs.io/en/latest/Singleton.html.

Reply:
The expectation is for users to not use this class directly and it should be exposed on the SDKs surface. I changed this to a private class and renamed them _Config and global_config to reduce the confusion.